app/src/test/java/co/touchlab/droidconandroid/shared/presenter/ConferenceViewModelTest.java

```diff
@@ -0,0 +1,91 @@
+package co.touchlab.droidconandroid.shared.presenter;
+
+import android.util.Pair;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import co.touchlab.doppl.testing.DopplContextDelegateTestRunner;
+import co.touchlab.droidconandroid.shared.data.AppPrefs;
+import co.touchlab.droidconandroid.shared.presenter.ConferenceDataViewModel;
+import io.reactivex.Observable;
+
+
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Mockito.only;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(DopplContextDelegateTestRunner.class)
+public class ConferenceViewModelTest
+{
+    @Mock
+    AppPrefs appPrefs;
+
+    @Mock
+    ConferenceDataViewModel.Host host;
+
+    ConferenceDataViewModel conferenceDataViewModel;
+    ConferenceDataViewModel.Factory factory;
+
+    @Before
+    public void setUpMockito()
+    {
+        MockitoAnnotations.initMocks(this);
+        factory = new ConferenceDataViewModel.Factory(true);
+        factory.appPrefs = appPrefs;
+        conferenceDataViewModel = factory.create(ConferenceDataViewModel.class);
+    }
+
+    @Test
+    public void whenDuplicateDates_ShouldUpdateHostOnce()
```

Comment:
These tests arent super useful because you are just testing that distinctUntilChanged works as advertised. Since you create the distinct observable in the test you are actually just testing your test code which doesnt really make sense. The only part of the production code that is tested here is the fact that the view model subscribes to the stream in register() and calls back to the host, which could be tested in a single test